crates/forge/src/main.rs

```diff
@@ -240,6 +241,34 @@ fn warn_if_snforge_std_not_compatible(scarb_metadata: &Metadata) -> Result<()> {
     Ok(())
 }
 
+fn maybe_save_tests_sierra_programs(
```

Comment:
Why here, not in `build_trace_data`?

Reply:
It's too late to do it in `build_trace_data`, you don't have `CrateLocation` there + I would have to pass around this weird path to test artifacts

Reply:
Can't we just put those into config? I think it is much less confusing for the reader than some weird saving procedure without context.

Reply:
We should run it from `maybe_save_execution_data` and add test_location to `TestCrateRunnable`